hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManagerImpl.java

```diff
@@ -1130,6 +1130,12 @@ private boolean isTableInGroup(TableName tableName, String groupName,
         if (region.isSplitParent()) {
           continue;
         }
+
+        // isSplitParent() and isSplit() is not always reliable. So for those scenarios, better to
```

Comment:
Why isSplitParent() and isSplit() is not always reliable?

Reply:
Hi Pankaj, thanks for reviewing. Found a pending TODO in `org.apache.hadoop.hbase.master.assignment.AssignmentManager#markRegionAsSplit` which explains why isSplit() is not reliable.
![image](https://user-images.githubusercontent.com/21206628/107346994-5cb29b80-6aeb-11eb-82b3-402583a971a1.png)
I have updated the MR by checking the region state instead of null server. 
I am also planning to open another JIRA for resolving that TODO if come up with a solution.